pandas/io/parsers.py

```diff
@@ -727,14 +700,78 @@ def parser_f(filepath_or_buffer,
-read_csv = Appender(_read_csv_doc)(read_csv)
+read_csv = Appender(_doc_read_csv_and_table.format(
+                    func_name='read_csv',
+                    summary=('Read a comma-separated values (csv) file '
+                             'into DataFrame.'),
+                    _default_sep="','")
+                    )(read_csv)
 
 read_table = _make_parser_function('read_table', default_sep='\t')
-read_table = Appender(_read_table_doc)(read_table)
+read_table = Appender(_doc_read_csv_and_table.format(
+                      func_name='read_table',
+                      summary="""Read general delimited file into DataFrame.
+
+.. deprecated:: 0.24.0
+Use :func:`pandas.read_csv` instead, passing ``sep='\\t'`` if necessary.""",
+                      _default_sep=r"'\\t' (tab-stop)")
+                      )(read_table)
+
 
+def read_fwf(filepath_or_buffer, colspecs='infer',
+             widths=None, **kwds):
+
+    r"""
+    Read a table of fixed-width formatted lines into DataFrame.
+
+    Also supports optionally iterating or breaking of the file
+    into chunks.
+
+    Additional help can be found in the `online docs for IO Tools
+    <http://pandas.pydata.org/pandas-docs/stable/io.html>`_.
+
+    Parameters
+    ----------
+    filepath_or_buffer : str, path object, or file-like object
+        Any valid string path is acceptable. The string could be a URL. Valid
+        URL schemes include http, ftp, s3, and file. For file URLs, a host is
+        expected. A local file could be: file://localhost/path/to/table.csv.
+
+        If you want to pass in a path object, pandas accepts either
+        ``pathlib.Path`` or ``py._path.local.LocalPath``.
+
+        By file-like object, we refer to objects with a ``read()`` method,
+        such as a file handler (e.g. via builtin ``open`` function)
+        or ``StringIO``.
+    colspecs : list of pairs (int, int) or 'infer'. optional
+        A list of pairs (tuples) giving the extents of the fixed-width
+        fields of each line as half-open intervals (i.e.,  [from, to[ ).
+        String value 'infer' can be used to instruct the parser to try
+        detecting the column specifications from the first 100 rows of
+        the data which are not being skipped via skiprows (default='infer').
+    widths : list of ints. optional
+        A list of field widths which can be used instead of 'colspecs' if
+        the intervals are contiguous.
+    **kwds : optional
+        Optional keyword arguments can be passed to ``TextFileReader``.
+
+    Returns
+    -------
+    DataFrame or TextParser
+        A comma-separated values (csv) file is returned as two-dimensional
+        data structure with labeled axes.
+
+    See Also
+    --------
+    to_csv : Write DataFrame to a comma-separated values (csv) file.
+    read_csv : Read a comma-separated values (csv) file into DataFrame.
+    read_fwf : Read a table of fixed-width formatted lines into DataFrame.
```

Comment:
Didn't see that before, can you remove this line? As this docstring is only used for `read_fwf`, there is no value in self-referencing itself.